### PR TITLE
fix(vdom): Fix issue causing multiple children (and as attrs) to not …

### LINKED ISSUE #830: Need to handle arrays for when specifying vdom children
We need to support something like:

```js
h('div', 
  h('span'),
  [h('span'), h('span')],
  [h('span'), h('span'), h('span')]
);
```

So this can work in JSX:

```js
<div>
  <span />
  {[0, 1].map(() => <span />)}
  {[0, 1, 2].map(() => <span />)}
</div>
```

### PR DESCRIPTION
**Pull request by treshugart**
…be rendered.

fixes #830

### CODE REVIEW

File: src/api/vdom.js [hunk 1]
```diff
@@ -321,6 +321,8 @@ export function element(tname, attrs, ...chren) {
       ch();
     } else if (ctype === 'string' || ctype === 'number') {
       newText(ch);
+    } else if (ctype === 'object') {
```
Comment (zzarcon) [inline src/api/vdom.js]: Is this safe enough to check if the type it's an Array? what about `Array.isArray`?

File: test/unit/vdom/builder.js [hunk 2]
```diff
@@ -79,5 +79,32 @@ describe('create()', () => {
...
+        rendered({ shadowRoot }) {
+          expect(shadowRoot.innerHTML).to.equal(
+            '<div>' +
+              '<span id="1"></span>' +
+              '<span id="2"></span>' +
+              '<span id="3"></span>' +
+              '<span id="4"></span>' +
+              '<span id="5"></span>' +
+              '<span id="6"></span>' +
+            '</div>'
```
Comment (zzarcon) [inline test/unit/vdom/builder.js]: Maybe is better to use a template literal for this?

```javascript
`<div>
  <span id="1"></span>
  <span id="2"></span>
  <span id="3"></span>
  <span id="4"></span>
  <span id="5"></span>
  <span id="6"></span>
</div>`
```

Comment (treshugart) [inline test/unit/vdom/builder.js]: Then the whitespace in the literal would fail the test (vdom doesn't generate whitespace).

**Comment by Hotell** [comment]
Hey folks! any update on this?

it is really inconvenient to call `jsx` transforms manually 

```jsx
const PleaseFixMe = () => {
  return [1,2,3].map( itm => <b>{itm}</b>() )
}
```

File: src/api/vdom.js (hunk 1 above)
Comment (Hotell) [inline src/api/vdom.js]: `Array.isArray` is the right way IMHO, `'object'` can be `null`,`Object`, `Array`

**Comment by treshugart** [comment]
@Hotell just restarting the builds now. I need this too :p

File: src/api/vdom.js (hunk 1 above)
Comment (treshugart) [inline src/api/vdom.js]: Yeah, not sure why I used `'object'`. Possibly because I was testing object literal traversal at some point? Will change.

**Pull request merged**